### PR TITLE
feat: indicate pre-init properties in device property browser

### LINKED ISSUE #375: Feature: indicate pre-init properties in device property browser
it's possible to filter by pre-init props in the device property browser, but it's not otherwise apparent _which_ of the properties are pre-init (without toggling).  These could have a small 🅟 icon or something next to them

### PR DESCRIPTION
**Pull request by fdrgsp**
tests failing because of https://github.com/hgrecco/pint/issues/2074.

update for #375 .

@tlambert03 instead of using a specific icon as you mentioned in #375 , what about changing the background color as we do for the read only props?

Screenshot 2024-11-06 at 8 34 57 AM

Screenshot 2024-11-06 at 8 35 17 AM

closes #375

### CODE REVIEW

**Comment by tlambert03** [comment]
that works for me.  did you happen to check how it looks on a dark theme?

**Comment by tlambert03** [comment]
oh well :) it's fine.  we've got enough dark theme issues anyway.  if we ever coordinate theming more generally we can address it then

**Comment by tlambert03** [comment]
however, if you're inclined, you could look into using a https://doc.qt.io/qt-6/qpalette.html#ColorRole-enum (which would pull from the palette) instead of hard-coding a color

File: src/pymmcore_widgets/device_properties/_device_property_table.py [hunk 1]
```diff
@@ -123,8 +123,16 @@ def _rebuild_table(self) -> None:
...
+
+        # Retrieve the palette
+        palette = self.palette()  # This pulls the current theme colors
+
+        read_only_color = palette.color(QPalette.ColorRole.Dark)
+        pre_init_color = palette.color(QPalette.ColorRole.Dark).darker(130)
+
         for i, prop in enumerate(props):
-            item = QTableWidgetItem(f"{prop.device}-{prop.name}")
+            extra = " (pre-init)" if prop.isPreInit() else ""
```
Comment (tlambert03) [inline src/pymmcore_widgets/device_properties/_device_property_table.py:134]: i'm afraid of the manual styling not working.  I think I would pretty much remove all the styling from this PR and just keep this line, but use:

```suggestion
            extra = " 🅿" if prop.isPreInit() else ""
```

Comment (fdrgsp) [reply src/pymmcore_widgets/device_properties/_device_property_table.py]: > i'm afraid of the manual styling not working in some (as of yet unseen) circumstance. 

Screenshot 2024-11-06 at 9 48 50 PM

**Pull request merged**